python/tests/operator/test_scalar_op.py

```diff
@@ -0,0 +1,371 @@
+    assert np.allclose(operator.add(2j, scop).to_matrix(), [2 + 2j])
+
+    assert np.allclose(operator.sub(scop, 2).to_matrix(), [0 + 0.j])
+    assert np.allclose(operator.sub(scop, 2.5).to_matrix(), [-0.5 + 0j])
+    assert np.allclose(operator.sub(scop, 2j).to_matrix(), [2 - 2j])
+    assert np.allclose(operator.sub(2, scop).to_matrix(), [0 + 0j])
+    assert np.allclose(operator.sub(2.5, scop).to_matrix(), [0.5 + 0j])
+    assert np.allclose(operator.sub(2j, scop).to_matrix(), [-2 + 2j])
+
+    assert np.allclose(operator.mul(scop, 2).to_matrix(), [4 + 0.j])
+    assert np.allclose(operator.mul(scop, 2.5).to_matrix(), [5 + 0j])
+    assert np.allclose(operator.mul(scop, 2j).to_matrix(), [0 + 4j])
+    assert np.allclose(operator.mul(2, scop).to_matrix(), [4 + 0j])
+    assert np.allclose(operator.mul(2.5, scop).to_matrix(), [5 + 0j])
+    assert np.allclose(operator.mul(2j, scop).to_matrix(), [0 + 4j])
+
+    assert np.allclose(operator.truediv(scop, 2).to_matrix(), [1 + 0.j])
+    assert np.allclose(operator.truediv(scop, 2.5).to_matrix(), [0.8 + 0j])
+    assert np.allclose(operator.truediv(scop, 2j).to_matrix(), [0 - 1j])
+    assert np.allclose(operator.truediv(2, scop).to_matrix(), [1 + 0j])
+    assert np.allclose(operator.truediv(2.5, scop).to_matrix(), [1.25 + 0j])
+    assert np.allclose(operator.truediv(2j, scop).to_matrix(), [0 + 1j])
+
+    assert np.allclose(operator.pow(scop, 2).to_matrix(), [4 + 0j])
+    assert np.allclose(
+        operator.pow(scop, 2.5).to_matrix(), [5.65685425 + 0j])
+    assert np.allclose(
+        operator.pow(scop, 2j).to_matrix(), [0.18345697 + 0.98302774j])
+    assert np.allclose(operator.pow(2, scop).to_matrix(), [4 + 0j])
+    assert np.allclose(operator.pow(2.5, scop).to_matrix(), [6.25 + 0j])
+    assert np.allclose(operator.pow(2j, scop).to_matrix(), [-4 + 0j])
+
+    so1 = ScalarOperator(lambda t: t)
+    assert so1.to_matrix(t=2.0) == 2.0
+    op1 = so1 * spin.x(0)
+    op2 = spin.x(0) * so1
+    op3 = so1 + spin.x(0)
+    op4 = spin.x(0) + so1
+    assert np.allclose(op1.to_matrix(t=2.0), [[0, 2], [2, 0]])
+    assert np.allclose(op2.to_matrix(t=2.0), [[0, 2], [2, 0]])
+    assert np.allclose(op3.to_matrix(t=2.0), [[2, 1], [1, 2]])
+    assert np.allclose(op4.to_matrix(t=2.0), [[2, 1], [1, 2]])
+    assert np.allclose(op1.to_matrix(t=1j), [[0, 1j], [1j, 0]])
+    assert np.allclose(op2.to_matrix(t=1j), [[0, 1j], [1j, 0]])
+    assert np.allclose(op3.to_matrix(t=1j), [[1j, 1], [1, 1j]])
+    assert np.allclose(op4.to_matrix(t=1j), [[1j, 1], [1, 1j]])
+
+    dims = {0: 2}
+    assert so1.to_matrix(dims, t=2.0) == 2.0
+    assert np.allclose(op1.to_matrix(dims, t=2.0), [[0, 2], [2, 0]])
+    assert np.allclose(op2.to_matrix(dims, t=2.0), [[0, 2], [2, 0]])
+    assert np.allclose(op3.to_matrix(dims, t=2.0), [[2, 1], [1, 2]])
+    assert np.allclose(op4.to_matrix(dims, t=2.0), [[2, 1], [1, 2]])
+
+
+# BREAKING CHANGE: setting the generator after construction is no longer supported
```

Comment:
I'll remove the commented out tests here before merging - I just thought it would be useful to see what exactly we no longer support.